### PR TITLE
Update publishdata.json to point to correct VS branch

### PR DESCRIPTION
**Pull request by dibarbet**

### CODE REVIEW

File: eng/config/PublishData.json [hunk 1]
```diff
@@ -208,7 +208,7 @@
       "version": "4.1.*",
       "packageFeeds": "default",
       "channels": [],
-      "vsBranch": "main",
```
Comment (ryzngard) [inline eng/config/PublishData.json:211]: This should target Roslyn's `release/dev17.1` branch instead of `main`

**Review by ryzngard — CHANGES_REQUESTED** [review]
Retarget this PR to our 17.1 P2 branch

**Comment by dibarbet** [comment]
@ryzngard I believe publishdata.json is always read from main (unless it has changed recently)
https://github.com/dotnet/roslyn/blob/875605dc866e4de86c2bce9430f19708003f8b48/azure-pipelines-official.yml#L323

**Review by JoeRobich — APPROVED** [review]

**Review by ryzngard — APPROVED** [review]

**Comment by JoeRobich** [comment]
@dibarbet should we also update main-vs-deps to not create draft PRs?

**Comment by dibarbet** [comment]
> @dibarbet should we also update main-vs-deps to not create draft PRs?

good call, updating

**Pull request merged**